cpp/include/cuspatial/experimental/allpairs_multipoint_equals_count.cuh

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <iterator>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute the number of multipoint pairs that are equal.
+ *
+ * Given two sets of multipoints, each represented by a range of `vec_2d<T>`s,
+ * computes the number of pairs of multipoints that are equal. Example:
+ *
+ * ```
+ * lhs: { {0, 0}, {1, 1}, {2, 2} }
+ * rhs: { {0, 0}, {1, 1}, {2, 2} }
+ * count: { 1, 1, 1 }
+ *
+ * lhs: { {0, 0} }
+ * rhs: { {0, 0}, {1, 1}, {2, 2}, {3, 3} }
```

Comment:
4. The example doesn't show that that it's not required for range to search from to be unsorted. (We will sort inside the function).

Reply:
done